### DISCUSSION TITLE
Help needed for overriding search in CrudPanel List

### DISCUSSION THREAD

**Discussion by fronbow** [Laravel-Backpack/community-forum#731] [Q&A (Help)] [answered]
Hi All

Does anyone know how to override the search function? 
I've been through the docs and think I understand how to use the search function, I'm just struggling with the applySearchTerm usage.
Is there any way of re-writing the sql query but still having it work with the DataTables functionality?

My use-case is:
I have a few tables that are searchable by first and last names, but now found out that some names of our clients are being spelled incorrectly by the staff.
e.g. Sean instead of Shaun
Jhon instead of John.

After a lot of testing (using both the php function and the mysql function), using SOUNDEX gives better results, so now I'm trying to work out how to alter the search query so that I can inject the soundex functionality. I've tried using the 'searchLogic' attribute of addColumn but this doesn't seem to give the correct results as it looks like it adds to whatever the existing query is.

Hope it makes sense?!

**Comment by phpfour** [comment]
Hey @fronbow, thanks for your question. 

If you'd like to override the search query completely, then you should have your own `search()` method in your Crud controller. 

<img width="600" src="https://github.com/Laravel-Backpack/community-forum/assets/171715/7aaaad38-dcce-48f0-9510-8bfd7e11d7dd" />

Just copy-paste the method from the original `vendor/backpack/crud/src/app/Http/Controllers/Operations/ListOperation.php` file and make changes to the below section to perform your own logic instead of calling the `applySearchTerm` method: 

```php
// if a search term was present
if ($search && $search['value'] ?? false) {
    // filter the results accordingly
    $this->crud->applySearchTerm($search['value']);
}
```

Hope this helps. Cheers!

**Reply by fronbow** [reply]
Hi @phpfour,

I've done this but I want to be able to override the sql query as the SearchTerm's can only be words; I need to compare `SOUNDEX`'s.
Is it possible to rewrite the query in the `search()` override? I feel more documentation is needed for this, like what is allowed in the above search block so that I can still let the function calculate the DataTable pagination?


Cheers

**Reply by pxpm** [reply, marked as answer]
@fronbow you have total control of the `search()` function return. 

You get the Crud query there but you don't need to use it. So you can do your completely custom query there. 

In the end just return the same `$this->crud->getEntriesAsJsonForDatatables($entries, $totalEntryCount, $filteredEntryCount, $start);` 

```php
$entries = //your db model collection
$totalEntryCount = // the count of the total entries in the db
$filteredEntryCount = // the count of the filteredEntries
$start = // how many entries where skipped
```

I suggest you `dd()` those variables in a "regular" crud panel, to have a better look on what that function receives as input, and you should be able to continue using the crud datatable pagination in your custom crud panel. 

Cheers

**Reply by fronbow** [reply]
Awesome, @pxpm , cheers for this, that's what I was hoping but it wasn't evident reading through the function.

As always, you're a star Sir :)
